scripts/download.pl

```diff
@@ -73,16 +73,25 @@ ()
 sub download_cmd($) {
 	my $url = shift;
 	my $have_curl = 0;
+	my $fn = shift;
+	my @mrs = @_;
+	my $murl = "'$url'";
 
+	my @chArray = ('a'..'z', 'A'..'Z', 0..9);
+	my $rfn = join '', map{ $chArray[int rand @chArray] } 0..9;
 	if (open CURL, '-|', 'curl', '--version') {
 		if (defined(my $line = readline CURL)) {
 			$have_curl = 1 if $line =~ /^curl /;
 		}
 		close CURL;
 	}
 
+	for my $el (@mrs) {
+		$murl = $murl." '$el/$fn'";
+	}
+
 	return $have_curl
-		? (qw(curl -f --connect-timeout 20 --retry 5 --location), $check_certificate ? '' : '--insecure', shellwords($ENV{CURL_OPTIONS} || ''), $url)
+		? ("aria2c --stderr $murl -c -x2 -s10 -j10 -k1M --server-stat-of=/dev/shm/spp --server-stat-if=/dev/shm/spp -d /dev/shm -o $rfn; cat /dev/shm/$rfn; rm /dev/shm/$rfn")
```

Comment:
This looks wrong. Probably you should introduce another variable `have_aria2c` and decide based on that to use aria2c (if found), fall back to curl (if found) and finally fall back to wget.

Reply:
hi dangowrt!
When I use 
```perl
my $have_aria2c = 0;
if (open ARIA2C, '-|', 'aria2c', '--version') {
	if (defined(my $line = readline ARIA2C)) {
		$have_aria2c = 1 if $line =~ /^aria2c /;
	}
	close ARIA2C;
}
```
variable `have_aria2c` always is `true`, I'm new to perl, I don't know what's going on.

Reply:
And I need to judge if aria2c found at line 93, 153, 303 and 305. I tried to define `my $have_aria2c = 0;` at line 31, but when I `make download -j1 V=sc`, terminal tells me `Bareword "have_aria2c" not allowed while "strict subs" in use at /home/bradford/workspace/openwrt/scripts/download.pl line 167.`. What should I do?